### PR TITLE
update CI to release and publish on taggings

### PR DESCRIPTION
**Pull request by robbkidd**
- [x] add CloudFormation template validation to CI tests
- [x] build the handlers and zipfile if all tests pass
- [x] publish to AWS
- [x] publish to GitHub Release
---
- [x] add CI user to repo
- ~turn off validate_templates job on forks, maybe?~ Punting. Not a big problem right now.

### CODE REVIEW

**Comment by robbkidd** [comment]
[This build in CI](https://app.circleci.com/pipelines/github/honeycombio/agentless-integrations-for-aws/69/workflows/2723ca64-8909-4425-8b88-b256c02fc984/jobs/79) shows the DRYRUN output of the publish_aws step before the job was set to run for-realz on taggings.

**Review by vreynolds — APPROVED** [review]
Non-blocking commentary 🥏

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,18 +31,79 @@ executors:
...
+  build:
     jobs:
-      - run_tests
-      # TODO build and deploy here
+      - go_test:
+          <<: *filters_always
+      - validate_templates:
+          <<: *filters_always
+          # Testing templates requires AWS API access; no changes are made
+          context: Honeycomb Secrets for Public Repos
```
[the remark this answers is not in the document]
Comment (robbkidd) [reply .circleci/config.yml:93]: Curses!

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,18 +31,79 @@ executors:
...
+  publish_github:
+    docker:
+      - image: cibuilds/github:0.13.0
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ./
+      - run:
+          name: "Draft GitHub Release"
+          command: ghr -draft -n ${CIRCLE_TAG} -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG} ./pkg/ingest-handlers.zip
```
[the remark this answers is not in the document]
Comment (robbkidd) [reply .circleci/config.yml:80]: ❗ Good catch, thank you, yes. Will add.

[the remark this answers is not in the document]
Comment (robbkidd) [reply .circleci/config.yml:80]: ✅   Bot user added.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (robbkidd) [reply .circleci/config.yml:93]: I cannot think of a way to solve this at the moment. Punting to solve this at some later date when we're getting so many outside contributions to the CloudFormation templates that it matters.

**Pull request merged**